Gulp was configured and installed

brianlee389 opened the pull request.

brianlee389 committed:
Gulp was configured

thomaspark commented on .gitignore:
```diff
@@ -0,0 +1,9 @@
+/node_modules/
```
This can be simplified:

`/node_modules/*`
`.DS_Store`

Also consider setting up a global gitignore file: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

thomaspark commented on README:
```diff
@@ -0,0 +1,43 @@
+Setting Up Gulp
```
Stray file?

thomaspark commented on gulpfile.js:
```diff
@@ -0,0 +1,49 @@
+var gulp = require('gulp');
```
It's good practice to order this. Something like gulp, gulp-\* (a to z), others (a to z)

thomaspark replied:
This file has a mix of tabs and spaces. Let's go with 2 spaces.

Also a mix of single and double quotes. Let's use double.

thomaspark commented on gulpfile.js:
```diff
@@ -0,0 +1,49 @@
+var rename = require('gulp-rename');
+var minifyCss = require('gulp-minify-css');
+
+var WORDPRESSURL = "http://localhost:8888/wordpresstest/wp-admin/edit.php?post_type=snowball";
```
This hard-coded URL doesn't work on my install, which is at http://localhost:8888/wordpress/

thomaspark commented on gulpfile.js:
```diff
@@ -0,0 +1,49 @@
+var WORDPRESSURL = "http://localhost:8888/wordpresstest/wp-admin/edit.php?post_type=snowball";
+
+gulp.task('minify', function () {
+    //var negateMinCSS = 'styles/!(*.min.css)'; for retrieving all non .min.css
```
Are these commented-out lines still needed?

thomaspark commented on gulpfile.js:
```diff
@@ -0,0 +1,49 @@
+    gulp.src('styles/*.css')
+        .pipe(minifyCss())
+        .pipe(rename(function(path) {
+            if (path.extname == ".css") {
```
Good to have checks, but is this needed given the glob pattern above?

thomaspark commented on gulpfile.js:
```diff
@@ -0,0 +1,49 @@
+});
+
+gulp.task('default', ['uglify', 'minify', 'browser-sync'], function () {
+    gulp.watch("scripts/*.js", ['uglify']).on('change', browserSync.reload);
```
You can pass an array of globs to `gulp.watch`. How about we create an array of our glob patterns, and reuse it in line 38 and here?

thomaspark commented on package.json:
```diff
@@ -0,0 +1,26 @@
+  "version": "1.0.0",
+  "description": "",
+  "main": "gulpfile.js",
+  "scripts": {
```
Not sure what `scripts` does. Can we drop it?

thomaspark commented on package.json:
```diff
@@ -0,0 +1,26 @@
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/brianlee389/snowball.git"
```
Set to main repo, license is MIT, homepage is http://snowball.openhtml.org

brianlee389 committed:
# This is a combination of 2 commits.
# The first commit's message is:

Setup gulp with minified files being used

# This is the 2nd commit message:

minifed fixedsticky files
